src/rdkafka_cgrp.c

```diff
@@ -6462,11 +6462,27 @@ void rd_kafka_cgrp_metadata_update_check(rd_kafka_cgrp_t *rkcg,
 
         rd_kafka_assert(NULL, thrd_is_current(rkcg->rkcg_rk->rk_thread));
 
+        if (rkcg->rkcg_group_protocol != RD_KAFKA_GROUP_PROTOCOL_GENERIC)
+                return;
+
         if (!rkcg->rkcg_subscription || rkcg->rkcg_subscription->cnt == 0)
                 return;
 
-        if (rkcg->rkcg_group_protocol != RD_KAFKA_GROUP_PROTOCOL_GENERIC)
+        if (!(rkcg->rkcg_flags & RD_KAFKA_CGRP_F_WILDCARD_SUBSCRIPTION) &&
+            rkcg->rkcg_join_state != RD_KAFKA_CGRP_JOIN_STATE_WAIT_METADATA &&
+            rkcg->rkcg_join_state != RD_KAFKA_CGRP_JOIN_STATE_STEADY) {
```

Comment:
For sure we know that in states RD_KAFKA_CGRP_JOIN_STATE_WAIT_INIT and RD_KAFKA_CGRP_JOIN_STATE_WAIT_JOIN it can cause the issue #4589 . I was also considering to put `rkcg_join_state >= RD_KAFKA_CGRP_JOIN_STATE_WAIT_METADATA` but doing these updates with a stable group seems more error proof.
In RD_KAFKA_CGRP_JOIN_STATE_WAIT_METADATA it's required because there are cases where it's waiting exactly this update.
For a regex subscription it's required in state INIT too, because it has to find the topics to include in the JoinGroup request, but in that case it doesn't have the problem with metadata cache.